### ISSUE TITLE
ephemeral volumes do not get cleaned up on host disappearance

### ISSUE THREAD

**Issue by erikh** [contiv/volplugin#16]
Right now, if the host disappears and docker loses state of the volume, we leak a volume that was intended to be deleted upon container/volume teardown inside of docker.

To resolve this, I propose we implement a scheduled jobs daemon that handles this and snapshots, and of course, can be used for more later. This `volsupervisor` could live on one host with a connection to the volmaster and etcd, and can be asynchronously managed with the rest of the cluster due to its tertiary purpose.

**Comment by erikh** [comment]
didn't mean to close this

**Comment by beni05** [comment]
This method requires shutting down the old instance before starting the migration and assigning its IP address to the new instance. This approach is not recommended unless the options above are unavailable.

**Comment by erikh** [comment]
I'm not quite sure what you mean. Can you elaborate? Are you talking about the volmaster or the volsupervisor?

**Comment by erikh** [comment]
this is done, finer points are going to be addressed in another issue
